src/figuresClasses.ts

```diff
@@ -2,10 +1,99 @@
+type Shape = string | 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Rectangle implements Figure {
+  shape: string = 'rectangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  private width: number;
+
+  private height: number;
+
+  constructor(color: 'red' | 'green' | 'blue', width: number, height: number) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('Width and height must be greater than zero!');
+    }
+    this.color = color;
+    this.width = width;
+    this.height = height;
+  }
+
+  getArea(): number {
+    const area = this.width * this.height;
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: string = 'circle';
+
+  color: 'red' | 'green' | 'blue';
+
+  private radius: number;
 
-export class Circle implements Figure {}
+  constructor(color: 'red' | 'green' | 'blue', radius: number) {
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than zero!');
+    }
+    this.color = color;
+    this.radius = radius;
+  }
 
-export class Rectangle implements Figure {}
+  getArea(): number {
+    const area = Math.PI * this.radius * this.radius;
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Triangle implements Figure {
+  shape: string = 'triangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  private a: number;
+
+  private b: number;
+
+  private c: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('The side cannot be less or equal to 0!');
+    }
+
+    if (Math.max(a, b, c) >= a + b + c - Math.max(a, b, c)) {
+      throw new Error(
+        "The longest side can't be more or equal to the sum of the rest sides!",
+      );
+    }
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
```

Comment:
[CODE KNOWLEDGE] - In 'Triangle', avoid manual property assignments in the constructor by using TypeScript's parameter properties. Add access modifiers to the constructor parameters to automatically create and assign class properties.